File: app/src/main/java/com/cpd/yuqing/db/vo/News.java
package com.cpd.yuqing.db.vo;

import android.database.Cursor;
import android.os.Parcel;
import android.os.Parcelable;

import java.util.List;

/**
 * Created by s21v on 2017/12/4.
 */

public class News implements Parcelable{
    private String news_id;		//新闻ID
    private int user_id; //用户ID
    private String channel_id;	//栏目ID
    private String homePageTitle;	//栏目页标题
    private String contentPageTitle;	//内容页标题
    private String pub_time;	//发布时间
    private String source;		//来源
    private String author;		//作者
    private String poster;		//责任编辑
    private String content;		//正文内容
    private String url;			//新闻的url链接
    private String picUrls;	//新闻中的图片链接
    private int favorite;   //是否收藏
    private int thumbUp;    //是否点赞

    public News(Cursor cursor) {
        this.news_id = cursor.getString(cursor.getColumnIndex("_id"));
        this.user_id = cursor.getInt(cursor.getColumnIndex("userId"));
        this.channel_id = cursor.getString(cursor.getColumnIndex("channelId"));
        this.homePageTitle = cursor.getString(cursor.getColumnIndex("homePageTitle"));
        this.contentPageTitle = cursor.getString(cursor.getColumnIndex("contentPageTitle"));
        this.pub_time = cursor.getString(cursor.getColumnIndex("pubTime"));
        this.source = cursor.getString(cursor.getColumnIndex("source"));
        this.author = cursor.getString(cursor.getColumnIndex("author"));
        this.poster = cursor.getString(cursor.getColumnIndex("poster"));
        this.content = cursor.getString(cursor.getColumnIndex("content"));
        this.url = cursor.getString(cursor.getColumnIndex("url"));
        this.picUrls = cursor.getString(cursor.getColumnIndex("picUrls"));
        this.favorite = cursor.getInt(cursor.getColumnIndex("favorite"));
        this.thumbUp = cursor.getInt(cursor.getColumnIndex("thumbUp"));
    }

    protected News(Parcel in) {
        news_id = in.readString();
        channel_id = in.readString();
        homePageTitle = in.readString();
        contentPageTitle = in.readString();
        pub_time = in.readString();
        source = in.readString();
        author = in.readString();
        poster = in.readString();
        content = in.readString();
        url = in.readString();
        picUrls = in.readString();
        favorite = in.readInt();
        thumbUp = in.readInt();
    }

    public static final Creator<News> CREATOR = new Creator<News>() {
        @Override
        public News createFromParcel(Parcel in) {
            return new News(in);
        }

        @Override
        public News[] newArray(int size) {
            return new News[size];
        }
    };

    public String getNews_id() {
        return news_id;
    }

    public String getChannel_id() {
        return channel_id;
    }

    public String getHomePageTitle() {
        return homePageTitle;
    }

    public void setHomePageTitle(String homePageTitle) {
        this.homePageTitle = homePageTitle;
    }

    public String getContentPageTitle() {
        return contentPageTitle;
    }

    public String getPub_time() {
        return pub_time;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getAuthor() {
        return author;
    }

    public String getPoster() {
        return poster;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getPicUrls() {
        return picUrls;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(news_id);
        dest.writeInt(user_id);
        dest.writeString(channel_id);
        dest.writeString(homePageTitle);
        dest.writeString(contentPageTitle);
        dest.writeString(pub_time);
        dest.writeString(source);
        dest.writeString(author);
        dest.writeString(poster);
        dest.writeString(content);
        dest.writeString(url);
        dest.writeString(picUrls);
        dest.writeInt(favorite);
        dest.writeInt(thumbUp);
    }

    @Override
    public String toString() {
        return "News{" +
                "news_id='" + news_id + '\'' +
                "user_id='" + user_id + '\'' +
                ", channel_id='" + channel_id + '\'' +
                ", homePageTitle='" + homePageTitle + '\'' +
                ", contentPageTitle='" + contentPageTitle + '\'' +
                ", pub_time='" + pub_time + '\'' +
                ", source='" + source + '\'' +
                ", author='" + author + '\'' +
                ", poster='" + poster + '\'' +
                ", content='" + content + '\'' +
                ", url='" + url + '\'' +
                ", picUrls=" + picUrls +
                '}';
    }

    public static String posterMix(final String poster) {
        return "责任编辑:"+poster;
    }
    public static String minPubTime(final String pubTime) { return pubTime.split("\\s+")[0];}
}

File: app/src/main/java/com/cpd/yuqing/data/News.kt
package com.cpd.yuqing.data

import android.os.Parcel
import android.os.Parcelable
import android.util.Log
import java.util.*
import kotlin.collections.ArrayList

/**
 * POJO : 新闻实体类
 * Created by s21v on 2017/6/27.
 * @param newsID : 新闻ID
 * @param channelID : 栏目ID
 * @param homePageTitle : 栏目页标题
 * @param contentPageTitle : 内容页标题
 * @param publishTime : 发布时间
 * @param source : 来源
 * @param author : 作者
 * @param poster : 责任编辑
 * @param pictureUrlList : 图片列表
 * @param content : 内容
 * @param contentUrlStr : 内容页链接
 * @param commentCount : 评论数
 */
data class News(val newsID: String="", val channelID: String="",
                var homePageTitle: String="", val contentPageTitle: String="",
                var publishTime: String="", var source: String="",
                val author: String="", val poster: String="",
                val content: String="", val contentUrlStr: String="",
                var commentCount: Int=0,
                var pictureUrlList: ArrayList<String>? = null) : Parcelable {

    fun getFormatCommentCount(): String = when {
        commentCount<1000 -> Integer.toString(commentCount)
        else -> "999+"
    }

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(newsID)
        dest.writeString(channelID)
        dest.writeString(homePageTitle)
        dest.writeString(contentPageTitle)
        dest.writeString(publishTime)
        dest.writeString(source)
        dest.writeString(author)
        dest.writeString(poster)
        dest.writeString(content)
        dest.writeString(contentUrlStr)
        dest.writeInt(commentCount)
        dest.writeStringList(pictureUrlList)
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<News> = object : Parcelable.Creator<News> {
            override fun createFromParcel(`in`: Parcel): News {
                val result = News(`in`.readString(), `in`.readString(), `in`.readString(),
                        `in`.readString(), `in`.readString(), `in`.readString(), `in`.readString(),
                        `in`.readString(), `in`.readString(), `in`.readString(), `in`.readInt())
                result.pictureUrlList = arrayListOf()
                `in`.readStringList(result.pictureUrlList)
                return result
            }

            override fun newArray(size: Int): Array<News> = newArray(size)
        }
    }
}
